src/main/java/com/example/ajaxonboardingproject/config/AppConfig.kt

```diff
@@ -1,20 +1,46 @@
 package com.example.ajaxonboardingproject.config
 
+import io.grpc.BindableService
+import io.grpc.Server
+import io.grpc.ServerBuilder
 import io.nats.client.Connection
 import io.nats.client.Nats
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Lazy
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
 
 @Configuration
-class AppConfig {
+class AppConfig(
     @Value("\${nats.connection.url}")
-    private lateinit var natsUrl: String
+    private val natsUrl: String,
+    @Value("\${spring.grpc.port}")
+    private val grpcPort: Int
+) {
+    @Autowired
+    @Lazy
+    private lateinit var bindableServices: List<BindableService>
 
     @Bean
     fun passwordEncoder(): BCryptPasswordEncoder = BCryptPasswordEncoder()
 
     @Bean
     fun natsConnection(): Connection = Nats.connect(natsUrl)
+
+    @Bean
+    fun getGrpcServer(): Server {
+        val grpcServer = ServerBuilder
+            .forPort(grpcPort)
+            .apply {
+                bindableServices.forEach { addService(it) }
+            }
+            .build()
+        Thread {
+            grpcServer.start()
+            grpcServer.awaitTermination()
+        }.start()
+        return grpcServer
```

Comment:
```suggestion
        val grpcServerBuilder = ServerBuilder
            .forPort(grpcPort)
            .apply {
                bindableServices.forEach { addService(it) }
            }
        return grpcServerBuilder.build().start()
```

will it work? If not, it's ok

Reply:
i've tried this approach before but it was not working in right way
but know it works great, thanks! 
(i suppose it had not working because of awaitTermination() method)
Done! ✅